node/src/components/consensus/highway_core/state.rs

```diff
@@ -254,8 +259,12 @@ impl<C: Context> State<C> {
     }
 
     /// Returns hash of vote that needs to be endorsed.
-    pub(crate) fn needs_endorsements(&self, _vote: &SignedWireVote<C>) -> Option<C::Hash> {
-        None
+    pub(crate) fn needs_endorsements(&self, vote: &SignedWireVote<C>) -> Option<C::Hash> {
+        vote.wire_vote
+            .endorsed
+            .iter()
+            .find(|hash| self.endorsements.contains_key(&hash))
```

Comment:
Did you mean
```suggestion
            .find(|hash| !self.endorsements.contains_key(&hash))
```
?
This should return the _missing_ dependency, should it?